docs/en/04_Changelogs/4.0.0.md

```diff
@@ -302,6 +302,26 @@ E.g.
 
 ## Upgrading
 
+### General upgrading
+
+In order to better support migration of legacy modules and code, an automatic class upgrade tool
```

Comment:
I don't think this gets to the USP quickly enough. People won't really understand what this tool is without context.

Maybe something like:

4.0 moves many classes to namespaces, and renames many methods. A tool is available to (semi-)automatically update your 3.x code.